mock/channels/index.html

```diff
@@ -1,73 +1,138 @@
+  <script>
+    onFdc3Ready().then(() => {
+      let firedOnce = false;
 
-        const fdc3ApiCalls = async (reverseFunctionCallOrder, contextBroadcasts) => {
-          await window.fdc3.leaveCurrentChannel();
-          switch (reverseFunctionCallOrder) {
-            case false:
-              await joinChannel();
-              broadcast(contextBroadcasts);
-              break;
-            case true:
-              await broadcast(contextBroadcasts);
-              joinChannel();
-              break;
-          }
+      const fdc3ApiCalls = async (context) => {
+        switch (context.reverseFunctionCallOrder) {
+          case false:
+            const testChannel = await joinChannel(context.joinAppChannel);
+            await broadcast(context, testChannel);
+            addCloseWindowListener(testChannel);
+            break;
+          case true:
+            await broadcast(context);
+            await joinChannel(context.joinAppChannel);
+            addCloseWindowListener();
+            break;
         }
+      }
 
-        //Channels app joins channel 1
-        const joinChannel = async () => {
-          const channels = await window.fdc3.getSystemChannels();
-          if (channels.length > 0) {
-            try {
-              await window.fdc3.joinChannel(channels[0].id);
-            } catch (ex) {
-              throw new Error("Error while calling joinChannel in app B: " + (ex.message ?? ex))
-            }
-          } else {
-            throw new Error("No system channels available for app B");
+      const addCloseWindowListener = async (testChannel) => {
+        if (testChannel === undefined) {
+          await window.fdc3.addContextListener(
+            "closeWindow",
+            () => {
+              closeFinsembleWindow();
+            });
+        } else {
+          await testChannel.addContextListener(
+            "closeWindow",
+            () => {
+              closeFinsembleWindow();
+            });
+        }
+      }
+
+      const joinChannel = async (useAppChannel) => {
+        if (useAppChannel) {
+          return await joinAppChannel();
+        } else {
+          await joinSystemChannel();
+        }
+      }
+
+      const broadcast = async (context, testChannel) => {
+        if (testChannel === undefined) {
+          await systemBroadcast(context.contextBroadcasts);
+        } else {
+          await appChannelBroadcast(context, testChannel);
+        }
+      }
+
+      //Channels app joins channel 1
+      const joinSystemChannel = async () => {
+        const channels = await window.fdc3.getSystemChannels();
+        if (channels.length > 0) {
+          try {
+            await window.fdc3.joinChannel(channels[0].id);
+            document.getElementById("context").innerHTML += `Joined user channel ${channels[0].id}/ `;
+          } catch (ex) {
+            throw new Error("Error while calling joinChannel in app B: " + (ex.message ?? ex))
```

Comment:
Does throwing errors like this achieve anything i.e. does it get displayed in the app? I would assume it doesn't do anything and might be better to add to the document?

Reply:
It doesn't. I was going to do this but forgot